pkg/core/bundle_unpack.go

```diff
@@ -155,6 +155,7 @@ func unpackBundleDescriptor(ctx context.Context, bundle *Bundle, publish bool) e
 type bundleEntriesRes struct {
 	bundleEntries model.BundleEntries
 	idx           uint64
+	id            string // the splitID when used in this context
```

Comment:
when we start talking about using a structure in multiple contexts, practically, what we have is a [sum type](https://github.com/fantasyland/daggy#daggytaggedsumtypename-constructors) type, also called an [algebraic data type](http://learnyouahaskell.com/making-our-own-types-and-typeclasses#algebraic-data-types), or [tagged union](https://en.wikipedia.org/wiki/Tagged_union), i guess is the wikipedia-official term.

i often find myself reaching for such a paradigm in golang only to recall that it doesn't provide tagged unions as a first-order concept, and [the standard reply](https://github.com/golang/go/issues/19412) about why not is that `interface`s are supposed to be used as functional replacements for tagged unions.

so i suggest doing just that:  create an interface type and use it instead of `bundleEntriesRes` when "context" in the sense of this comment cannot be determined from variables' types.  at the simplest, it could provide a [catamorphism](https://github.com/ransomw/kvsjs/blob/master/src/dsl.js#L26) (i.e. fan-out) from the `interface` type to the original `bundleEntriesRes` and an additional type similar to `bundleEntriesRes` except in contexts where `splitID` is used, although there are other interface designs available.

what i'm saying here, most generally, is, "let's try to encode structure of the source first, then use comments for clarification and generating prose-like documentation."

Reply:
With this PR, I am trying to reuse what I can from existing bundle stuff.
A lot of such interesting theoretical points indeed come up to mind when doing so.
The practical outcome is what I did. The comment stuff is left behind so I may refactor this later on.